pkg/featureflag/feature_flags.go

```diff
@@ -30,10 +31,29 @@ const FeatureFlagEnabledDefault = true
 // increments a telemetry counter for any feature flag that is denied and logs
 // the feature and category that was denied.
 func CheckEnabled(
-	ctx context.Context, s *settings.BoolSetting, sv *settings.Values, featureName string,
+	ctx context.Context, config Config, s *settings.BoolSetting, featureName string,
 ) error {
+	sv := config.SV()
+
 	if enabled := s.Get(sv); !enabled {
+		// Increment telemetry counter for feature flag denial.
 		telemetry.Inc(sqltelemetry.FeatureDeniedByFeatureFlagCounter)
+
+		// Increment metrics counter for feature flag denial.
+		if config.GetFeatureFlagMetrics() == nil {
+			log.Warningf(
+				ctx,
+				"executorConfig.FeatureFlagMetrics is uninitiated; feature %s was attempted but disabled via cluster settings",
```

Comment:
Would there be a way to call out the associated cluster setting here (idk if that information is provided by `featureName` or elsewhere)?

Reply:
At the moment no, the `featureName` is not linked to the associated cluster setting. I can see how that would be useful. I know that @vy-ton is considering the best way to document which cluster settings may block which statements (as in the future there may be multiple categories that one statement can fall under).